.gitlab-ci.yml

```diff
@@ -1,11 +1,40 @@
 ---
+image: node:10-alpine
+
+variables:
+# https://docs.gitlab.com/ce/ci/variables/README.html
+  app_version: $CI_COMMIT_REF_SLUG-$CI_COMMIT_SHA
+  registry_image: $CI_REGISTRY_IMAGE
+  branch_or_tag: $CI_COMMIT_REF_NAME
+  DOCKER_HOST: tcp://localhost:2375
+
 stages:
-- "Sample build step"
+  - info
+  - "build and push base image"
+  - "build and push base-dependencies image"
+  - "build and push tremplin images"
 
-sample_build:
-  stage: "Sample build step"
-  image: node:current-slim
-  variables:
-    DOCKER_HOST: tcp://localhost:2375
+info:
+  stage: info
+  image: docker:latest
   script:
-  - npm install
+    - docker --version
+
+include:
+  - local: '/scripts/build/.gitlab-ci/build.gitlab-ci.yml'
+
+
+# build-and-push-tremplin-frontend-image:
+#   stage: "Build & Push Docker Image"
+#   image: docker:latest
+#   variables:
+#     DOCKER_HOST: tcp://localhost:2375
+#   services:
+#   - docker:dind
+#   script:
+#   - cd packages/app
+#   - docker login  $CI_REGISTRY -u gitlab-ci-token -p $CI_JOB_TOKEN
+#   - docker build -t "$CI_REGISTRY_IMAGE/frontend:latest" .
```

Comment:
"Latest is not a version" !
Est-ce que tagger avec le nom de la branche ou le SHA du commit ferai le job ? La CI fournie plein de variables qui devraient faire le job. Sur des builds en // exécutés depuis différentes branches, est-ce que ça risque pas de péter d'avoir uniquement `latest` ? Typiquement dans un cas où sur une branche il y a une MàJ des libs / dépendance et pas dans l'autre ?

Reply:
Cette partie a été supprimée.Dans les scripts, les images sont marquées comme `latest` que sur la branche `master`, sinon elles sont taggées avec comme ca `$CI_COMMIT_REF_SLUG-$CI_COMMIT_SHA`.